src/controllers/tipoDocumento.js

```
@@ -0,0 +1,106 @@
+/** @type {import('mongoose').Model} */
+const TipoDocumento = require('../models/tipodocumento');
+const mensaje = require('../helpers/mensajes');
+const { NotFoundException, BadRequestException } = require('../errors');
+
+const create = async (req, res, next) => {
+  const { body } = req;
+  try {
+    const tipoDoc = await TipoDocumento.findOne({
+      descripcion: body.descripcion,
+    });
+    if (tipoDoc)
+      throw new BadRequestException(
+        'El tipo de documento ya existe',
+        'tipodocumento',
+      );
+
+    const tDoc = await TipoDocumento.create(body);
+    res.status(201).json({
+      message: mensaje.registroCorrecto,
+      data: tDoc,
+    });
+  } catch (err) {
+    next(err);
+  }
+};
+
+const getAll = async (req, res, next) => {
+  try {
+    const tDoc = await TipoDocumento.find();
+    res.status(200).json({
+      message: mensaje.listadoCorrecto,
+      data: tDoc,
+    });
+  } catch (err) {
+    next(err);
+  }
+};
+
+const getOne = async (req, res, next) => {
+  const { id } = req.params;
+  try {
+    const tDoc = await TipoDocumento.findById(id);
+    if (!tDoc)
+      throw new NotFoundException(
+        mensaje.tipoDocumentoNoExiste,
+        'tipodocumento',
+      );
+    res.status(200).json({
+      message: mensaje.objetoEcontrado,
+      data: tDoc,
+    });
+  } catch (err) {
+    next(err);

```

No t necessary to change, but sometimes it is good to `throw` an error, instead of simple letting the logic continue